src/PCSC.Iso7816/IsoReader.cs

```diff
@@ -299,6 +299,11 @@ public class IsoReader : IIsoReader
 
                 // In case there is more data available.
                 le = responseApdu.SW2;
+                if (responseApdu.SW1 == (byte)SW1Code.Normal) {
+                    // add the last ResponseAPDU to the Response object
+                    response.Add(responseApdu);

```

Comment:
I do not understand how this will help.

At line 236 & 237 the last `responseApdu` is already added to the response.